server/delay_server/src/irodsDelayServer.cpp

```diff
@@ -676,6 +677,51 @@ namespace
     } // make_delay_queue_query_processor
 } // anonymous namespace
 
+bool is_current_server_leader(void) {
+    std::string hostname;
+    char hn[HOST_NAME_MAX + 1]{};
+
+    if (gethostname(hn, sizeof(hn)) != 0) {
+        logger::delay_server::warn("is_current_server_leader: Failed to retrieve local server's hostname.");
+        return false;
+    }
+
+    if (const auto hn_resolved = resolve_hostname(hn, hostname_resolution_scheme::match_preferred); hn_resolved) {
+        hostname = *hn_resolved;
+    }
+    else {
+        hostname = std::string{hn};
+    }
+    auto conn = get_new_connection(std::nullopt);
+    std::string result;
+    {
+        GridConfigurationInput input{};
+        std::strcpy(input.name_space, "delay_server");
+        std::strcpy(input.option_name, "leader");
+
+        GridConfigurationOutput* output{};
+        irods::at_scope_exit free_output{[&output] { std::free(output); }};
+
+        if (const auto ec = rc_get_grid_configuration_value(static_cast<RcComm*>(conn), &input, &output); ec != 0) {
+            logger::delay_server::warn("is_current_server_leader: Failed to retriever leader config option.");
+            return false;
+        }
+
+        result = output->option_value;
+    }
+
+    std::string leader;
+    if (const auto hn_resolved = resolve_hostname(result, hostname_resolution_scheme::match_preferred); hn_resolved) {
+        leader = *hn_resolved;
+    }
+    else {
+        leader = result;
+    }
```

Comment:
Do we need to resolve the leader value?   It's just coming from the database... so, that's a setting, not something that is coming from the OS that needs to have bumpers applied...?

Does this behave if we always set `leader = result`?

Reply:
The resolve_hostname function is using what used to be the hosts_config.json file to obtain a valid iRODS hostname.

I don't think we can remove that call because we can't guarantee the host resolution information is always available on every server.

Reply:
To add onto this, I mostly lifted this from rodsServer...if the rationale is the same as it is there, this should be fine?

Reply:
Agreed. Marking as resolved.